addons/binding/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/communication/AbstractResponse.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.digiplex.communication;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Abstract response
+ *
+ * @author Robert Michalak - Initial contribution
+ *
+ */
+@NonNullByDefault
+public abstract class AbstractResponse implements DigiplexResponse {
```

Comment:
Make this an immutable class (no getters, all fields public final)

Reply:
It is an immutable class actually. I may add `final` to `success` field to make it more clear, but I don't see the reason to make it public and remove getters. 

If this is the way immutable classes are expressed in this project, then we should also modify all inheritors of `AbstractResponse` to keep the same pattern everywhere.

Reply:
The classes are more concise, easier to review, and immutable class members cannot be changed anyway. Yes using the same pattern everywhere would reduce the submitted size by some extend and would be appreciated.

Reply:
OK, I can do that. 

I'm curious though is it a new direction going forward for immutable classes? Cause I can see many examples in the code that do not follow your guidelines and I'm a bit reluctant to introduce a new pattern, which looks more like an exception than a standard currently.

Reply:
The old object oriented programming pattern requested to use getters. openHAB just turned 10 years, so yes not everything is consistent. 

In all my bindings and in all my binding reviews an immutable class is without getters.